specs/sequencer-app.md

```diff
@@ -41,143 +42,306 @@ idea of "messages", where every transaction can contain multiple messages. The
+`ed25519_consensus::VerificationKey` (i.e. the public key).
 
-## ABCI block lifecycle
+## Actions
+
+The following is an exhaustive list of all user-accessible actions available
+to be submitted as of Sequencer v1.0.0-rc.1.
+
+### Core Protocol Actions
+
+* `Transfer`: represents a value transfer action between two accounts. It consists
+of the following fields:
+
+  | **Field** | **Type** | **Description** |
+  | --------- | -------- | ----------- |
+  | to        | `Address`| The account to transfer to. The "from" address is assumed|
+  ||| to be the signer of the transaction. |
+  | amount    | `uint128`| The amount to transfer. |
+  | asset     | `string` | The asset to transfer. |
+  | fee_asset | `string` | The asset used to pay for the action's fees. |
+
+* `RollupDataSubmission`: a transaction ordered by the sequencer, whose destination
+is another chain. It consists of the following:
+
+  | **Field** | **Type** | **Description** |
+  | --------- | -------- | ----------- |
+  | rollup_id | `RollupId`| ID of the destination chain. |
+  | data      | `bytes`  | The opaque transaction data. |
+  | fee_asset | `string` | The asset used to pay for the action's fees. |
+
+### Bridge Actions
+
+These actions deal with transfering funds to/from a bridge account to be used on
+a rollup.
+
+* `InitBridgeAccount`: initializes a bridge account for the given rollup on the
+sequencer chain. The signer of the transaction is the owner of the bridge account,
+and the only actor authorized to transfer out of the account. It is made up of
+the following fields:
+
+  | **Field** | **Type** | **Description** |
+  | --------- | -------- | ----------- |
+  | rollup_id | `RollupId` | The rollup ID with which to register the bridge account.|
+  | asset     | `string`   | The asset ID that will be accepted by the account.|
+  | fee_asset | `string`   | The asset with which to pay fees. |
+  | sudo_address | `Address` | The address which has authority over the bridge|
+  ||| account. If empty, assigned to the signer. |
+  | withdrawer_address | `Address` | The address which is allowed to withdraw funds|
+  ||| from the bridge account. If empty, assigned to the signer.|
+
+* `BridgeLock`: transfers funds from a sequencer account to a bridge account.
+It is effectively similar to `Transfer`. Upon execution of a bridge lock action,
+a `Deposit` event will be emitted, containing the information of the transfer.
+
+  | **Field** | **Type** | **Description** |
+  | --------- | -------- | ----------- |
+  | to        | `Address`| The bridge account to transfer to. The "from" address|
+  ||| is assumed to be the signer of the transaction. |
+  | amount    | `uint128`| The amount to transfer. |
+  | asset     | `string` | The asset to transfer. |
+  | fee_asset | `string` | The asset used to pay for the action's fees. |
+  | destination_chain_address | `string` | The address on the destination chain|
+  ||| which will receive the bridged funds. |
```

Comment:
the description for `destination_chain_address` is split weird here

Reply:
This is to appease the linter, while still keeping it looking (hopefully) decent. Ex:

| field | type | description |
| ----  | ---- | ------ |
| fee_asset | `string` | The asset used to pay for the action's fees. |
| destination_chain_address | `string` | The address on the destination chain|
||| which will receive the bridged funds. |